vector/src/main/java/im/vector/app/features/voicebroadcast/usecase/StartVoiceBroadcastUseCase.kt

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2022 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.voicebroadcast.usecase
+
+import im.vector.app.features.voicebroadcast.STATE_ROOM_VOICE_BROADCAST_INFO
+import im.vector.app.features.voicebroadcast.model.MessageVoiceBroadcastInfoContent
+import im.vector.app.features.voicebroadcast.model.VoiceBroadcastState
+import org.matrix.android.sdk.api.query.QueryStringValue
+import org.matrix.android.sdk.api.session.Session
+import org.matrix.android.sdk.api.session.events.model.toContent
+import org.matrix.android.sdk.api.session.events.model.toModel
+import org.matrix.android.sdk.api.session.room.Room
+import timber.log.Timber
+import javax.inject.Inject
+
+class StartVoiceBroadcastUseCase @Inject constructor(
+        private val session: Session,
+) {
+
+    suspend fun execute(roomId: String) {
+        val room = session.roomService().getRoom(roomId) ?: return
+
+        Timber.d("## StartVoiceBroadcastUseCase: Start voice broadcast requested")
+
+        val lastVoiceBroadcastEvent = room.stateService().getStateEvent(STATE_ROOM_VOICE_BROADCAST_INFO, QueryStringValue.Equals(session.myUserId))
+        when (val voiceBroadcastState = lastVoiceBroadcastEvent?.content.toModel<MessageVoiceBroadcastInfoContent>()?.voiceBroadcastState) {
+            VoiceBroadcastState.STOPPED,
+            null -> startVoiceBroadcast(room)
+            else -> Timber.d("## StartVoiceBroadcastUseCase: Cannot start voice broadcast: currentState=$voiceBroadcastState")
+        }
+    }
+
+    private suspend fun startVoiceBroadcast(room: Room) {
+        Timber.d("## StartVoiceBroadcastUseCase: Send new voice broadcast info state event")
+        room.stateService().sendStateEvent(
```

Comment:
This method can throw (no network for instance). Maybe protect by a try/catch. (applicable to the other similar calls)

Reply:
I didn't find a better way than catching the whole execute function: https://github.com/vector-im/element-android/pull/7273/commits/b06c2afdf0af3c5b87b0ffea47511464a1af2a56

Reply:
@bmarty I let you review this commit to resolve or not this conversation here

Reply:
ok, this is fine.